api/server/handlers/registry/get_token.go

```diff
@@ -402,35 +402,77 @@ func NewRegistryGetACRTokenHandler(
 }
 
 func (c *RegistryGetACRTokenHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	proj, _ := r.Context().Value(types.ProjectScope).(*models.Project)
+	ctx, span := telemetry.NewSpan(r.Context(), "serve-acr-token")
+	defer span.End()
+
+	proj, _ := ctx.Value(types.ProjectScope).(*models.Project)
+
+	telemetry.WithAttributes(span, telemetry.AttributeKV{Key: "project-id", Value: proj.ID})
 
 	// list registries and find one that matches the region
 	regs, err := c.Repo().Registry().ListRegistriesByProjectID(proj.ID)
 	if err != nil {
-		c.HandleAPIError(w, r, apierrors.NewErrInternal(err))
+		err := telemetry.Error(ctx, span, err, "error getting registries by project id")
+		c.HandleAPIError(w, r, apierrors.NewErrPassThroughToClient(err, http.StatusInternalServerError))
 		return
 	}
 
 	var token string
 	var expiresAt *time.Time
 
 	for _, reg := range regs {
-		if reg.AzureIntegrationID != 0 && strings.Contains(reg.URL, "azurecr.io") {
-			_reg := registry.Registry(*reg)
-			username, pw, err := _reg.GetACRCredentials(c.Repo())
-			if err != nil {
-				c.HandleAPIError(w, r, apierrors.NewErrInternal(err))
-				continue
-			}
+		if strings.Contains(reg.URL, "azurecr.io") {
+			telemetry.WithAttributes(span, telemetry.AttributeKV{Key: "registry-name", Value: reg.Name})
 
-			token = base64.StdEncoding.EncodeToString([]byte(string(username) + ":" + string(pw)))
+			if proj.CapiProvisionerEnabled {
+				telemetry.WithAttributes(span, telemetry.AttributeKV{Key: "capi-provisioned", Value: true})
 
-			// we'll just set an arbitrary 30-day expiry time (this is not enforced)
-			timeExpires := time.Now().Add(30 * 24 * 3600 * time.Second)
-			expiresAt = &timeExpires
+				tokenReq := connect.NewRequest(&porterv1.TokenForRegistryRequest{
+					ProjectId:   int64(proj.ID),
+					RegistryUri: reg.URL,
+				})
+				tokenResp, err := c.Config().ClusterControlPlaneClient.TokenForRegistry(ctx, tokenReq)
+				if err != nil {
+					err := telemetry.Error(ctx, span, err, "error getting token response from ccp")
+					c.HandleAPIError(w, r, apierrors.NewErrPassThroughToClient(err, http.StatusInternalServerError))
+					return
+				}
+
+				if tokenResp.Msg == nil || tokenResp.Msg.Token == "" {
+					err := telemetry.Error(ctx, span, err, "no token found in response")
```

Comment:
watch out for this spans where err isn't explicitly set 
```suggestion
					err := telemetry.Error(ctx, span, nil, "no token found in response")
```